examples/CMakeLists.txt

```diff
@@ -48,3 +48,15 @@ IF(DEAL_II_COMPONENT_EXAMPLES)
 
   MESSAGE(STATUS "Setting up examples - Done")
 ENDIF()
+
+#
+# Add a dummy target to make files known to IDEs like qtcreator
+#
+
+FILE(GLOB _misc
+  ${CMAKE_CURRENT_SOURCE_DIR}/step-*/*.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/step-*/*.h
+)
+
+ADD_LIBRARY(example_misc_files EXCLUDE_FROM_ALL ${_misc})
+SET_TARGET_PROPERTIES(example_misc_files PROPERTIES LINKER_LANGUAGE C)
```

Comment:
Yes, I can see this leading to all sorts of awkwardness if anything ever tries to build such a library. For one, it has many `main()` functions.